src/main/java/com/android/volley/toolbox/HttpHeaderParser.java

```diff
@@ -135,7 +135,15 @@ public static long parseDateAsEpoch(String dateStr) {
             return newRfc1123Formatter().parse(dateStr).getTime();
         } catch (ParseException e) {
             // Date in invalid format, fallback to 0
-            VolleyLog.e(e, "Unable to parse dateStr: %s, falling back to 0", dateStr);
+            // If the value is either "0" or "-1" we only log to verbose,
+            // these values are pretty common and cause log spam.
+            String message = "Unable to parse dateStr: %s, falling back to 0";
+            if (dateStr.equals("0") || dateStr.equals("-1")) {
```

Comment:
nit: this looks okay in this particular case, but I would prefer:

"0".equals(dateStr) || "1".equals(dateStr)

because this more clearly cannot cause a NullPointerException where one was impossible before. As currently written, you would have to reason about whether parse(dateStr) would have thrown an NullPointerException before this code would have been reached. The answer depends on the implementation of SimpleDateFormat, since that behavior is undocumented, and it looks like it would generally throw a NullPointerException, but better not to rely on that and just gracefully accept a null dateStr here since nothing else is enforcing it clearly.